openmc/mesh.py

```diff
@@ -1511,6 +1517,12 @@ class SphericalMesh(StructuredMesh):
     indices : Iterable of tuple
         An iterable of mesh indices for each mesh element, e.g. [(1, 1, 1),
         (2, 1, 1), ...]
+    lower_left : Iterable of float
+        The lower-left corner of the structured mesh. If only two coordinate
+        are given, it is assumed that the mesh is an x-y mesh.
+    upper_right : Iterable of float
+        The upper-right corner of the structured mesh. If only two coordinate
+        are given, it is assumed that the mesh is an x-y mesh.
 
```

Comment:
I wonder if we need bounding_box in the  something in the doc string to explain that the bounding box is not a minimum / smallest bounding as in some cases it might be larger than necessary. I think this is fine but I wonder if we just want to mention this to the user in the doc string for bounding_box. I'm not entirely sure so best to see what the others think

Reply:
@shimwell 
You're saying you think more verbose documentation would be helpful? Ok. 

I would think the next step would be to do analogous pull request on the rectilinear mesh. Then abstract bounding_box property to the StructuredMesh class using duck typing.

Reply:
To add on here, there should be a description of `bounding_box` in this docstring.